xl-v1.go

```diff
@@ -429,8 +429,26 @@ func (xl XL) StatFile(volume, path string) (FileInfo, error) {
 		return FileInfo{}, errInvalidArgument
 	}
 
+	readLock := true
```

Comment:
@vadmeste I think, it can be simplified by having `xl.rLockNS()` and `xl.rUnlockNS()`.

The idea behind not using [rwMutex](https://github.com/minio/minio/blob/xl-layer/xl-v1-namespace.go#L29) directly is [nameSpaceLockMap](https://github.com/minio/minio/blob/xl-layer/xl-v1.go#L48) may have a stale lock entry forever.  This is avoided by tracking a [counter](https://github.com/minio/minio/blob/xl-layer/xl-v1-namespace.go#L30) of the rwMutex usage and when its zero after [unlock](https://github.com/minio/minio/blob/xl-layer/xl-v1.go#L79), the name space lock won't be stored back in the map.

Please share your thoughts to simplify this.